From 18700c16ed427d134ff61df5300855db47159ba1 Mon Sep 17 00:00:00 2001
From: Lim Siew Hoon <siew.hoon.lim@intel.com>
Date: Wed, 2 Aug 2023 12:09:27 +0800
Subject: [PATCH 164/232] wayland: Export the _get_type() functions

Add the macro before the declaration so the types are properly exported,
this makes it possible to do dynamic casting and checks.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/3858>
Backport from upstream:
https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/16158df5b20bbe3e4a9ff668cb82ca2412a71636

Signed-off-by: Lim Siew Hoon <siew.hoon.lim@intel.com>
---
 gst-libs/gst/wayland/gstwlbuffer.h          | 2 ++
 gst-libs/gst/wayland/gstwldisplay.h         | 2 ++
 gst-libs/gst/wayland/gstwlshmallocator.h    | 2 ++
 gst-libs/gst/wayland/gstwlvideobufferpool.h | 2 ++
 gst-libs/gst/wayland/gstwlwindow.h          | 2 ++
 5 files changed, 10 insertions(+)

diff --git a/gst-libs/gst/wayland/gstwlbuffer.h b/gst-libs/gst/wayland/gstwlbuffer.h
index 07034919b..08db591dd 100644
--- a/gst-libs/gst/wayland/gstwlbuffer.h
+++ b/gst-libs/gst/wayland/gstwlbuffer.h
@@ -25,6 +25,8 @@
 G_BEGIN_DECLS
 
 #define GST_TYPE_WL_BUFFER gst_wl_buffer_get_type ()
+
+GST_WL_API
 G_DECLARE_FINAL_TYPE (GstWlBuffer, gst_wl_buffer, GST, WL_BUFFER, GObject);
 
 struct _GstWlBuffer
diff --git a/gst-libs/gst/wayland/gstwldisplay.h b/gst-libs/gst/wayland/gstwldisplay.h
index 957d73e93..5093fa33b 100644
--- a/gst-libs/gst/wayland/gstwldisplay.h
+++ b/gst-libs/gst/wayland/gstwldisplay.h
@@ -28,6 +28,8 @@
 G_BEGIN_DECLS
 
 #define GST_TYPE_WL_DISPLAY (gst_wl_display_get_type ())
+
+GST_WL_API
 G_DECLARE_FINAL_TYPE (GstWlDisplay, gst_wl_display, GST, WL_DISPLAY, GObject);
 
 struct _GstWlDisplay
diff --git a/gst-libs/gst/wayland/gstwlshmallocator.h b/gst-libs/gst/wayland/gstwlshmallocator.h
index 0e7e15da9..edf60a9fc 100644
--- a/gst-libs/gst/wayland/gstwlshmallocator.h
+++ b/gst-libs/gst/wayland/gstwlshmallocator.h
@@ -30,6 +30,8 @@
 G_BEGIN_DECLS
 
 #define GST_TYPE_WL_SHM_ALLOCATOR (gst_wl_shm_allocator_get_type ())
+
+GST_WL_API
 G_DECLARE_FINAL_TYPE (GstWlShmAllocator, gst_wl_shm_allocator, GST, WL_SHM_ALLOCATOR, GstFdAllocator);
 
 #define GST_ALLOCATOR_WL_SHM "wl_shm"
diff --git a/gst-libs/gst/wayland/gstwlvideobufferpool.h b/gst-libs/gst/wayland/gstwlvideobufferpool.h
index fe04186c6..780fb6ff1 100644
--- a/gst-libs/gst/wayland/gstwlvideobufferpool.h
+++ b/gst-libs/gst/wayland/gstwlvideobufferpool.h
@@ -34,6 +34,8 @@ G_BEGIN_DECLS
  * pool also support GstDRMDumbAllocator. */
 
 #define GST_TYPE_WL_VIDEO_BUFFER_POOL (gst_wl_video_buffer_pool_get_type ())
+
+GST_WL_API
 G_DECLARE_FINAL_TYPE (GstWlVideoBufferPool, gst_wl_video_buffer_pool, GST, WL_VIDEO_BUFFER_POOL, GstVideoBufferPool);
 
 GST_WL_API
diff --git a/gst-libs/gst/wayland/gstwlwindow.h b/gst-libs/gst/wayland/gstwlwindow.h
index 06c40014f..f3b338202 100644
--- a/gst-libs/gst/wayland/gstwlwindow.h
+++ b/gst-libs/gst/wayland/gstwlwindow.h
@@ -26,6 +26,8 @@
 G_BEGIN_DECLS
 
 #define GST_TYPE_WL_WINDOW (gst_wl_window_get_type ())
+
+GST_WL_API
 G_DECLARE_FINAL_TYPE (GstWlWindow, gst_wl_window, GST, WL_WINDOW, GObject);
 
 struct _GstWlWindow
-- 
2.40.1

